0.download_data/download_figshare.py

```diff
@@ -0,0 +1,52 @@
+import os
+import pathlib
+import shutil
+import urllib.request
+import zipfile
+
+
+def download_figshare(
+    figshare_id: str,
+    output_file: pathlib.Path,
+    output_dir: pathlib.Path,
+    metadata_dir: pathlib.Path,
+    figshare_url: str = "https://ndownloader.figshare.com/files/",
+):
+    """
+    Download the provided figshare resource and extract the files from Figshare. Extract the downloaded
+    zip file into a folder and then seperate the metadata for each plate into a seperate metadata folder.
+
+    Attributes
+    ----------
+    figshare_id : str
+        string of numbers that corresponds to the figshare identifier
+    output_file : pathlib.Path
+        the location and file name of the downloaded zip file
+    output_dir : pathlib.Path
+        path to directory for images
+    metadata_dir : pathlib.Path
+        path to directory for metadata
+    figshare_url: str, default "https://ndownloader.figshare.com/files/"
+        the location of where the figshare id is stored
+    """
+    # access the url and download the zip file from figshare containing files for plate (images + metadata)
+    urllib.request.urlretrieve(f"{figshare_url}/{figshare_id}", output_file)
+
+    # find the zip file and then extract it into the specific folder
+    with zipfile.ZipFile(output_file, "r") as zip_files:
```

Comment:
this won't always be necessary - consider adding `unzip_files=True` as a function argument.

Reply:
Added, it is interesting why my files are zipped while other projects on Figshare do not download as zip files